main.bal

```diff
@@ -1,56 +1,128 @@
 import ballerina/io;
 import ballerina/log;
-// import ballerina/test;
 import ballerina/tcp;
-// import ballerina/lang.'string;
-import ballerina/regex;
 
-
-//bind the service to the port
 service on new tcp:Listener(3000) {
 
-    remote function onConnect(tcp:Caller caller)
-                              returns tcp:ConnectionService {
+    remote function onConnect(tcp:Caller caller) 
+                            returns tcp:ConnectionService {
         io:println("Client connected to the server: ", caller.remotePort);
         return new Service();
     }
 }
 
 service class Service {
-//once the content is received from the client, this method is invoked
 
     remote function onBytes(tcp:Caller caller, readonly & byte[] data) 
         returns tcp:Error? {
-        // io:println(string:fromBytes(data));  
+        error? e = self.handleInput(caller, data);
+
+        if e is error {
+            return;
+        }
+    }
 
-        string|error firststring = string:fromBytes(data);   
+    byte[] buffer = [];
+    byte[] buffer2 = [];
+    int? expectedLen = ();
+    int? lengthOne = ();
+    int? lengthTwo = ();
+    int x = 0;
```

Comment:
If we are going with one letter, I like `n` better than `x` here. I think eventually we'll have to get rid of this altogether. outgoing message should be picked by looking at incoming massage, not by the count, but lets do that later after we fix other suggestions.